src/Propel/Common/Util/PathTrait.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/**
+ * MIT License. This file is part of the Propel package.
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Propel\Common\Util;
+
+use RuntimeException;
+
+trait PathTrait
+{
+    /**
+     * @var string|null
+     */
+    protected ?string $path = null;
+
+    /**
+     * Template paths are by convention in
+     * - templates/
+     * - besides src/ folder (which is required to detect root path)
+     *
+     * Note:
+     * - Propel/Generator/ prefix is removed from the path
+     *
+     * Examples:
+     * - Behavior/BehaviorName/
+     * - Builder/Om/
+     * - Command/
+     * - Manager/
+     *
+     * @param string $path
+     *
+     * @throws \RuntimeException
+     *
+     * @return string
+     */
+    protected function getTemplatePath(string $path): string
+    {
+        if ($this->path) {
```

Comment:
I have my concern regarding this code. We have a param that can change a return value, but we will always return the result of the first call of this method. it mb ok for the current usecases, but IMO this is unexpected and not transparent

Reply:
Should we avoid this lookup you mean?
It could fail if we have nested objects doing getTemplatePath() from different points of extension, sure.

Reply:
I mean this method should not store result in a property